### PR TITLE
(DOCSP-23333): Asymmetric Sync docs for Swift SDK

### PR DESCRIPTION
**Pull request by dacharyc**
### Jira

- https://jira.mongodb.org/browse/DOCSP-23333

### Staged Changes (Requires MongoDB Corp SSO)

- [Model Data/Define an Object Model](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-23333/sdk/swift/model-data/define-model/object-models/#define-an-asymmetric-object): New "Define an Asymmetric Object" section
- [CRUD/Create](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-23333/sdk/swift/crud/create/#create-an-asymmetric-object): New "Create an Asymmetric Object" section
- [Sync Data/Configure & Open a Synced Realm](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-23333/sdk/swift/sync/configure-and-open-a-synced-realm/#open-a-synced-realm-for-flexible-sync): Update note to specify Asymmetric Sync does not require a subscription
- [Sync Data/Manage Flexible Sync Subscriptions](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-23333/sdk/swift/sync/flexible-sync/#overview): Add note that Asymmetric Sync does not require a subscription
- [Sync Data/Stream Data to Atlas](https://docs-mongodbcom-staging.corp.mongodb.com/realm/docsworker-xlarge/DOCSP-23333/sdk/swift/sync/stream-data-to-atlas/): New task-oriented rollup page with some overview content for context, and a procedure putting all the pieces together and calling out some of the specific differences between Asymmetric and bidirectional Sync

### Review Guidelines

[REVIEWING.md](https://github.com/mongodb/docs-realm/blob/master/REVIEWING.md)

### CODE REVIEW

File: source/sdk/swift/model-data/define-model/object-models.txt [hunk 1]
```diff
@@ -654,3 +654,36 @@ property in several ways:
...
+.. _swift-define-asymmetric-object:
+
+Define an Asymmetric Object
+--------------------------
+
+.. versionadded:: 10.29.0
+
+You can use Flexible Sync to sync an object unidirectionally
+from your device to the database linked to your Atlas App Services App. 
+Define an asymmetric object by deriving from :swift-sdk:`AsymmetricObject
```
Comment (mongodben) [inline source/sdk/swift/model-data/define-model/object-models.txt:667]: is "deriving" a swift-specific technical term? if not, consider rephrasing as it's not standard term, as far as i know.

File: source/sdk/swift/model-data/define-model/object-models.txt [hunk 2]
```diff
@@ -654,3 +654,36 @@ property in several ways:
...
+.. literalinclude:: /examples/generated/code/start/AsymmetricSync.snippet.asymmetric-model.swift
+   :language: swift
+
+``AsymmetricObjects`` broadly support the same property types as ``Object``, 
+with a few exceptions:
+
+- Asymmetric objects can only link to embedded objects
+
+   - ``Object`` and ``List<Object>`` properties are not supported 
+   - ``EmbeddedObject`` and ``List<EmbeddedObject>`` are supported
```
Comment (mongodben) [inline source/sdk/swift/model-data/define-model/object-models.txt:678]: i think this isn't supposed to be nested list

File: source/sdk/swift/model-data/define-model/object-models.txt [hunk 3]
```diff
@@ -654,3 +654,36 @@ property in several ways:
...
+
+You cannot link an ``AsymmetricObject`` from within an ``Object``. Doing so
```
Comment (mongodben) [inline source/sdk/swift/model-data/define-model/object-models.txt:680]:
```suggestion
You cannot link to an ``AsymmetricObject`` from within an ``Object``. Doing so 
```

File: source/sdk/swift/model-data/define-model/object-models.txt [hunk 4]
```diff
@@ -654,3 +654,36 @@ property in several ways:
...
+
+- Asymmetric objects can only link to embedded objects
+
+   - ``Object`` and ``List<Object>`` properties are not supported 
+   - ``EmbeddedObject`` and ``List<EmbeddedObject>`` are supported
+
+You cannot link an ``AsymmetricObject`` from within an ``Object``. Doing so 
+throws an error.
+
+Asymmetric objects cannot be added to or removed from a realm, or queried.
```
Comment (mongodben) [inline source/sdk/swift/model-data/define-model/object-models.txt:683]: i believe use of commas in this sentence isn't grammatically correct.

File: source/sdk/swift/sync/configure-and-open-a-synced-realm.txt [hunk 5]
```diff
@@ -174,9 +174,13 @@ to open a synced realm.
 
 .. important:: Flexible Sync Requires a Subscription
 
-   You can't use a Flexible Sync realm until you add at least one subscription.
-   To learn how to add subscriptions, see: :ref:`<ios-sync-add-subscription>`.
-   
+   If your app uses regular Device Sync, you can't use a Flexible Sync
```
Comment (mongodben) [inline source/sdk/swift/sync/configure-and-open-a-synced-realm.txt:177]: what is "regular device sync"? i'd rephrase this to as it was, and then lead into how this doesn't apply Asymmetric Sync by starting `Unless you use Asymmetric Sync, in which case...`

File: source/sdk/swift/sync/flexible-sync.txt [hunk 6]
```diff
@@ -33,11 +33,16 @@ To use Flexible Sync in an iOS client:
 You can add, update, and remove query subscriptions to determine which data 
 syncs to the client device.
 
+**Note**: Flexible Sync subscriptions only apply to using Atlas Device Sync 
+with regular Realm Objects. If your app uses Asymmetric Sync to unidirectionally
```
Comment (mongodben) [inline source/sdk/swift/sync/flexible-sync.txt:37]: rather than "regular Realm Objects" could you link to the class, as you do for `AsymmetricObjects`

File: source/sdk/swift/sync/flexible-sync.txt [hunk 7]
```diff
@@ -33,11 +33,16 @@ To use Flexible Sync in an iOS client:
...
+sync :swift-sdk:`AsymmetricObjects <Extensions/AsymmetricObject.html>` via an 
+Atlas App Services App, you do not create subscriptions for those objects.
```
Comment (mongodben) [inline source/sdk/swift/sync/flexible-sync.txt:39]: so asymmetric objects just sync automatically w/o having to do the whole sync config? 

if this is the case, i think it's worth explicitly stating this, and explaining the behavior in a small section about Asymmetric Sync

File: source/sdk/swift/model-data/define-model/object-models.txt (hunk 1 above)
Comment (dacharyc) [reply source/sdk/swift/model-data/define-model/object-models.txt:667]: It's a pretty common term in describing Swift inheritance; it's an alternative to "child".

File: source/sdk/swift/model-data/define-model/object-models.txt (hunk 2 above)
Comment (dacharyc) [reply source/sdk/swift/model-data/define-model/object-models.txt:678]: It is, actually, but I have updated the spacing as that extra line in there was causing a Snooty build issue.

File: source/sdk/swift/model-data/define-model/object-models.txt (hunk 4 above)
Comment (dacharyc) [reply source/sdk/swift/model-data/define-model/object-models.txt:683]: Good call - rewording to remove passive voice and making this a bullet list instead.

File: source/sdk/swift/sync/configure-and-open-a-synced-realm.txt (hunk 5 above)
Comment (dacharyc) [reply source/sdk/swift/sync/configure-and-open-a-synced-realm.txt:177]: The implication I am trying to make is that there's your bog standard Device Sync/Flexible Sync, which requires a subscription, and then Asymmetric Sync is a very special subset that doesn't comply with the same rules as the bidirectional version of Device Sync/Flex Sync. I don't want to state or imply that "Flexible Sync" always requires a subscription, because that is now not the case with this special subset of Flex Sync.

Maybe:
> If your app uses bidirectional Device Sync, you can't use a Flex...

What do you think of that?

File: source/sdk/swift/sync/flexible-sync.txt (hunk 7 above)
Comment (dacharyc) [reply source/sdk/swift/sync/flexible-sync.txt:39]: Not exactly. You still need a Flexible Sync config, you just don't use subscriptions with it. That's why I don't have a section about it on this page, because this page is about Managing Flexible Sync Subscriptions and you explicitly can't use a subscription with Asymmetric Sync.

File: source/sdk/swift/sync/configure-and-open-a-synced-realm.txt (hunk 5 above)
Comment (mongodben) [reply source/sdk/swift/sync/configure-and-open-a-synced-realm.txt:177]: i like inclusion of word "bidirectional". also think it's worth noting that this is the standard. maybe: 

> ...bidirectional, standard Device Sync ...

File: source/sdk/swift/sync/flexible-sync.txt (hunk 6 above)
Comment (mongodben) [reply source/sdk/swift/sync/flexible-sync.txt:37]: i think "regular" is too vague. maybe just delete the first sentence?

File: source/sdk/swift/sync/flexible-sync.txt (hunk 7 above)
Comment (mongodben) [reply source/sdk/swift/sync/flexible-sync.txt:39]: ok, riding on the above comment thread about removing sentence "Flexible Sync subscriptions...", perhaps you also remove the "**Note**" to make clearer that this is an important detail to setting up asymmetric sync.

File: source/sdk/swift/sync/flexible-sync.txt (hunk 6 above)
Comment (mongodben) [inline source/sdk/swift/sync/flexible-sync.txt:37]: i think you should link the term "Asymmetric Sync" to https://www.mongodb.com/docs/atlas/app-services/sync/data-access-patterns/flexible-sync/#optimize-sync-with-asymmetric-sync--optional-

File: source/sdk/swift/model-data/define-model/object-models.txt [hunk 8]
```diff
@@ -654,3 +654,41 @@ property in several ways:
...
   projection. 
+
+.. _swift-define-asymmetric-object:
+
+Define an Asymmetric Object
+--------------------------
+
+.. versionadded:: 10.29.0
+
+You can use Flexible Sync to sync an object unidirectionally
```
Comment (mongodben) [inline source/sdk/swift/model-data/define-model/object-models.txt:665]: somewhere toward the beginning of this paragraph it's worth explicitly mentioning asymmetric sync and linking to https://www.mongodb.com/docs/atlas/app-services/sync/data-access-patterns/flexible-sync/#optimize-sync-with-asymmetric-sync--optional-

File: source/sdk/swift/crud/create.txt [hunk 9]
```diff
@@ -350,6 +350,27 @@ you must check the type before you do anything with the value.
...
+Create an Asymmetric Object
+---------------------------
+
+.. versionadded:: 10.29.0
+
+You can create an AsymmetricObject using :swift-sdk:`create(_ type:, value:)
+<Structs/Realm.html#/s:10RealmSwift0A0V6create_5valueyxm_yptSo0aB16AsymmetricObjectCRbzlF>`.
+When you create an AsymmetricObject, it syncs unidirectionally to the Atlas 
+database linked to your Atlas App Services App. You cannot access an
+AsymmetricObject locally, add it to or remove it from a realm, or query for it.
```
Comment (mongodben) [inline source/sdk/swift/crud/create.txt:364]: sry missed mentioning before, but prob also worth linking to Asymmetric Sync docs here as well, as you just did on the other pages.

**Review by mongodben — APPROVED** [review]
lgtm, 1 remaining item to address in new comment

File: examples/ios/Examples/AsymmetricSync.swift [hunk 10]
```diff
@@ -0,0 +1,96 @@
...
+                                       value: [ "_id": ObjectId.generate(),
+                                                "deviceId": "WX1278UIT",
+                                                "temperatureInFahrenheit": 66.7,
+                                                "barometricPressureInHg": 29.65,
+                                                "windSpeedInMph": 2
+                                                ])
+            }
+            // :snippet-end:
+            // Add a delay to give the document time to sync
+            sleep(10)
```
Comment (dianaafanador3) [inline examples/ios/Examples/AsymmetricSync.swift:78]: mmm probably we don't want to suggest the user to use sleep, what we want is to check if the document until the document is updated. This is only if the user wants to check this, as the use case for asymmetric objects is to not be queried or read locally after created

File: source/examples/generated/code/start/AsymmetricSync.snippet.asymmetric-model.swift [hunk 11]
```diff
@@ -0,0 +1,15 @@
+class WeatherSensor: AsymmetricObject {
+    @Persisted(primaryKey: true) var _id: ObjectId
+    @Persisted var deviceId: String
+    @Persisted var temperatureInFahrenheit: Float
+    @Persisted var barometricPressureInHg: Float
+    @Persisted var windSpeedInMph: Int
+    
+    convenience init(deviceId: String, temperatureInFahrenheit: Float, barometricPressureInHg: Float, windSpeedInMph: Int) {
```
Comment (dianaafanador3) [inline source/examples/generated/code/start/AsymmetricSync.snippet.asymmetric-model.swift:8]: No need to add a convenience init if is not  been used

File: source/sdk/swift/crud/create.txt [hunk 12]
```diff
@@ -350,6 +350,28 @@ you must check the type before you do anything with the value.
...
+<Structs/Realm.html#/s:10RealmSwift0A0V6create_5valueyxm_yptSo0aB16AsymmetricObjectCRbzlF>`.
+When you create an AsymmetricObject, it syncs unidirectionally via 
+:ref:`Asymmetric Sync <optimize-asymmetric-sync>` to the Atlas 
+database linked to your Atlas App Services App. You cannot access an
+AsymmetricObject locally, add it to or remove it from a realm, or query for it.
+
+.. literalinclude:: /examples/generated/code/start/AsymmetricSync.snippet.create-asymmetric-object.swift
+   :language: swift
+
+You can create AsymmetricObjects for a realm initialized with a :swift-sdk:`Flexible Sync configuration
```
Comment (dianaafanador3) [inline source/sdk/swift/crud/create.txt:370]: You can **only**

File: source/sdk/swift/model-data/define-model/object-models.txt [hunk 13]
```diff
@@ -654,3 +654,42 @@ property in several ways:
...
+
+Define an Asymmetric Object
+--------------------------
+
+.. versionadded:: 10.29.0
+
+If your app uses Flexible Sync, you can use :ref:`Asymmetric Sync 
+<optimize-asymmetric-sync>` to sync an object unidirectionally
+from your device to the database linked to your Atlas App Services App. 
+Define an asymmetric object by deriving from :swift-sdk:`AsymmetricObject
```
Comment (dianaafanador3) [inline source/sdk/swift/model-data/define-model/object-models.txt:668]:
```suggestion
Define an asymmetric object by inheriting from :swift-sdk:`AsymmetricObject 
```

File: source/sdk/swift/sync/stream-data-to-atlas.txt [hunk 14]
```diff
@@ -0,0 +1,103 @@
...
+         Generally, Realm expects that you do not mix ``Object`` and 
+         ``AsymmetricObject`` types within a project. If your project does
+         contain both types, you must explicitly :ref:`pass a subset of 
+         classes in your Realm configuration 
+         <ios-provide-a-subset-of-classes-to-a-realm>` each time you open 
+         a Realm. 
+         
+         You cannot query an AsymmetricObject or write it to a local
+         realm, so AsymmetricObjects are incompatible with bi-directional
+         Flexible Sync or local Realm use. Automatic schema discovery means
```
Comment (dianaafanador3) [inline source/sdk/swift/sync/stream-data-to-atlas.txt:88]: Also incompatible with PBS Sync Realms

File: examples/ios/Examples/AsymmetricSync.swift (hunk 10 above)
Comment (dacharyc) [reply examples/ios/Examples/AsymmetricSync.swift:78]: Yeah, the `snippet-end` in line 76 is where the docs code excerpt stops; the sleep and following stuff is just for me to check that I am using this successfully and it's uploading to the backend in our tests. I did not provide an example for the user to check this, since the use case for this is heavy insert workloads and verifying the upload doesn't seem to fit with that.
